Análise exploratória dados fitbit

In [2]:
import pandas as pd

In [3]:
d_activity = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/dailyActivity_merged.csv')
heartrate = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/heartrate_seconds_merged.csv')
h_calories = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/hourlyCalories_merged.csv')
h_intensity = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/hourlyIntensities_merged.csv')
m_calories = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/minuteCaloriesNarrow_merged.csv')
m_intensity = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/minuteIntensitiesNarrow_merged.csv')
m_sleep = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/minuteSleep_merged.csv')
weightlog = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/weightLogInfo_merged.csv')

### Análise exploratória d_activity


In [4]:
d_activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 457 entries, 0 to 456
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        457 non-null    int64  
 1   ActivityDate              457 non-null    object 
 2   TotalSteps                457 non-null    int64  
 3   TotalDistance             457 non-null    float64
 4   TrackerDistance           457 non-null    float64
 5   LoggedActivitiesDistance  457 non-null    float64
 6   VeryActiveDistance        457 non-null    float64
 7   ModeratelyActiveDistance  457 non-null    float64
 8   LightActiveDistance       457 non-null    float64
 9   SedentaryActiveDistance   457 non-null    float64
 10  VeryActiveMinutes         457 non-null    int64  
 11  FairlyActiveMinutes       457 non-null    int64  
 12  LightlyActiveMinutes      457 non-null    int64  
 13  SedentaryMinutes          457 non-null    int64  
 14  Calories  

In [5]:
d_activity['Id'].nunique() #35 ids nos dados de d_activity

35

In [17]:
d_activity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    3/25/2016       11004           7.11             7.11   
1  1503960366    3/26/2016       17609          11.55            11.55   
2  1503960366    3/27/2016       12736           8.53             8.53   
3  1503960366    3/28/2016       13231           8.93             8.93   
4  1503960366    3/29/2016       12041           7.85             7.85   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                2.57                      0.46   
1                       0.0                6.92                      0.73   
2                       0.0                4.66                      0.16   
3                       0.0                3.19                      0.79   
4                       0.0                2.16                      1.09   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 4.07                      0.0                 33   
1                 3.91                      0.0                 89   
2                 3.71                      0.0                 56   
3                 4.95                      0.0                 39   
4                 4.61                      0.0                 28   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   12                   205               804      1819  
1                   17                   274               588      2154  
2                    5                   268               605      1944  
3                   20                   224              1080      1932  
4                   28                   243               763      1886

### Análise Total steps

#### Estatística dos dados de passos

In [6]:
d_activity['TotalSteps'].median() #5986.0
d_activity['TotalSteps'].mean() #6546.5
d_activity['TotalSteps'].mode() #0  61 linhas desse dataframe não tem nenhuma entrada de exercício, ou passos. 
#396 entradas com input de passos

0    0
Name: TotalSteps, dtype: int64

#### Análise de passos max e mean por id

In [7]:
pd.pivot_table(d_activity, values = 'TotalSteps', index= 'Id', aggfunc= ('mean', 'max')).round(2)
#apenas uma pessoa não fez input de passos, média e máximo de passos parece sem erros

max      mean
Id                         
1503960366  17609  11640.53
1624580081  18464   4226.26
1644430081  20237   9274.80
1844505072   6847   3640.58
1927972279   5662   2180.83
2022484408  18247  12174.92
2026352035   6679   3392.75
2320127002  11230   3138.42
2347167796  16081   9800.07
2873212765  14103   6636.58
2891001357   4126    773.62
3372868164   8827   6127.50
3977333714  15425   8663.92
4020332650  12483   5776.59
4057192912   8433   1887.22
4319703577  10946   7820.58
4388161847      0      0.00
4445114986   6911   4293.07
4558609924   8828   5785.25
4702921684  14002   7943.40
5553957443  15444   8354.75
5577150313  14873   8608.18
6117666160  19658   8248.60
6290855005   9548   1618.10
6391747486   9766   1336.89
6775888955  13987   5559.00
6962181067  18952  12639.71
7007744171  16806  12260.33
7086361926  14604   6103.92
8053475328  25701  14844.36
8253242879  12026   2389.92
8378563200  14179   8135.25
8583815059   6835   3045.50
8792009665   8452   3094.92
8877689391  28497  17417.08

In [8]:
d_activity[d_activity['TotalSteps']== 0] #dados nulos? é possível entender que essas pessoas só usam o fitbit em dias de exercício?

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
51   1844505072     4/4/2016           0            0.0              0.0   
52   1844505072     4/5/2016           0            0.0              0.0   
59   1844505072    4/12/2016           0            0.0              0.0   
96   2320127002     4/1/2016           0            0.0              0.0   
97   2320127002     4/2/2016           0            0.0              0.0   
..          ...          ...         ...            ...              ...   
402  8253242879     4/2/2016           0            0.0              0.0   
403  8253242879     4/3/2016           0            0.0              0.0   
404  8253242879     4/4/2016           0            0.0              0.0   
405  8253242879     4/5/2016           0            0.0              0.0   
412  8253242879    4/12/2016           0            0.0              0.0   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
51                        0.0                 0.0                       0.0   
52                        0.0                 0.0                       0.0   
59                        0.0                 0.0                       0.0   
96                        0.0                 0.0                       0.0   
97                        0.0                 0.0                       0.0   
..                        ...                 ...                       ...   
402                       0.0                 0.0                       0.0   
403                       0.0                 0.0                       0.0   
404                       0.0                 0.0                       0.0   
405                       0.0                 0.0                       0.0   
412                       0.0                 0.0                       0.0   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
51                   0.0                      0.0                  0   
52                   0.0                      0.0                  0   
59                   0.0                      0.0                  0   
96                   0.0                      0.0                  0   
97                   0.0                      0.0                  0   
..                   ...                      ...                ...   
402                  0.0                      0.0                  0   
403                  0.0                      0.0                  0   
404                  0.0                      0.0                  0   
405                  0.0                      0.0                  0   
412                  0.0                      0.0                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
51                     0                     0               669      1347  
52                     0                     0              1440      1347  
59                     0                     0               426       399  
96                     0                     0              1440      1324  
97                     0                     0              1440      1324  
..                   ...                   ...               ...       ...  
402                    0                     0              1440      1429  
403                    0                     0              1440      1429  
404                    0                     0              1440      1429  
405                    0                     0              1440      1429  
412                    0                     0              1440         0  

[61 rows x 15 columns]

In [ ]:
# pesquisando entradas dos ids com steps nulos - 14 pessoas com dados nulos de passos - sabendo que apenas um desses Ids (4388161847) não tem nenhum input de passos
null_steps_ids = d_activity.loc[d_activity['TotalSteps']== 0, 'Id'].drop_duplicates()
null_steps_ids

51     1844505072
96     2320127002
122    2347167796
134    2873212765
135    2891001357
197    4057192912
240    4319703577
241    4388161847
290    4702921684
316    6117666160
326    6290855005
334    6391747486
346    6775888955
401    8253242879
Name: Id, dtype: int64

### Insights sobre steps até aqui:
* 34 Ids tem input de passos, sem dados discrepantes
* um Id não inputou passos nenhuma vez
* 14 ids tem dias com 0 passos, como não é viável passar 1 dia sem dar passos, pode-se supor que são pessoas que não usam o fit bit em dias sem exercício (e nessa amostra, são minoria)

### Análise exploratória heartrate

In [10]:
heartrate.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1154681 entries, 0 to 1154680
Data columns (total 3 columns):
 #   Column  Non-Null Count    Dtype 
---  ------  --------------    ----- 
 0   Id      1154681 non-null  int64 
 1   Time    1154681 non-null  object
 2   Value   1154681 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 26.4+ MB


In [18]:
heartrate.head()

Id                 Time  Value
0  2022484408  4/1/2016 7:54:00 AM     93
1  2022484408  4/1/2016 7:54:05 AM     91
2  2022484408  4/1/2016 7:54:10 AM     96
3  2022484408  4/1/2016 7:54:15 AM     98
4  2022484408  4/1/2016 7:54:20 AM    100

In [11]:
heartrate['Id'].nunique() #14 ids com heartrate, menos que a metade dos dados de passos

14

In [12]:
pd.pivot_table(heartrate, values = 'Value', index= 'Id', aggfunc= ['mean', 'max', 'min']).round(2).sort_values(by= ('mean', 'Value'), ascending= False)

mean   max   min
            Value Value Value
Id                           
6775888955  97.74   185    59
7007744171  89.93   158    56
8877689391  87.60   179    47
6391747486  84.12   133    49
6117666160  83.49   130    52
4020332650  81.90   151    46
2022484408  81.69   182    46
6962181067  80.69   176    48
4558609924  79.55   158    50
8792009665  76.33   163    45
2347167796  76.07   182    51
5553957443  69.42   152    48
2026352035  65.78    80    57
5577150313  65.00   176    36

### Insights até aqui
* apenas 14 ids tem input de heart rate
* destes 14, 4 ids estão em comum com a lista de pessoas que teorizei que só usam durante exercício (6775888955 , 6391747486, 6117666160 , 2347167796), o que ainda pode fazer sentido, a valor médio de batimentos é alto em 3 delas - pode indicar novamente fitbit apenas quando faz exercício

### Análise exploratória hourly calories
* 34 ids tem input de calorias

In [13]:
h_calories.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24084 entries, 0 to 24083
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Id            24084 non-null  int64 
 1   ActivityHour  24084 non-null  object
 2   Calories      24084 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 564.6+ KB


In [33]:
h_calories.head()

Id           ActivityHour  Calories ActivityDay
0  1503960366  3/12/2016 12:00:00 AM        48   3/12/2016
1  1503960366   3/12/2016 1:00:00 AM        48   3/12/2016
2  1503960366   3/12/2016 2:00:00 AM        48   3/12/2016
3  1503960366   3/12/2016 3:00:00 AM        48   3/12/2016
4  1503960366   3/12/2016 4:00:00 AM        48   3/12/2016

In [34]:
h_calories['Id'].nunique() #34 ids com input de calorias (gastas)

34

In [ ]:
h_calories['ActivityDay'] = h_calories['ActivityHour'].str.extract(r'^([^\s]+)')

daily_calories = pd.pivot_table(h_calories, values= 'Calories', index= ('Id', 'ActivityDay'), aggfunc= 'sum')
daily_calories

Calories
Id         ActivityDay          
1503960366 3/12/2016        2228
           3/13/2016        2100
           3/14/2016        1830
           3/15/2016        2111
           3/16/2016        1967
...                          ...
8877689391 4/5/2016         2832
           4/6/2016         4034
           4/7/2016         2941
           4/8/2016         4191
           4/9/2016         3835

[1021 rows x 1 columns]

In [32]:
avg_daily_calories = pd.pivot_table(daily_calories, values = 'Calories', index='Id', aggfunc= 'mean').round(2).sort_values(by='Calories', ascending= False)
avg_daily_calories

Calories
Id                  
8877689391   3435.25
5577150313   3434.71
8378563200   3338.84
4702921684   2957.23
1644430081   2956.90
8053475328   2941.59
6290855005   2652.27
4020332650   2633.62
6775888955   2631.03
7007744171   2609.22
1927972279   2580.94
2022484408   2568.50
7086361926   2292.34
8583815059   2286.64
6117666160   2270.83
4445114986   2192.34
6962181067   2170.50
4319703577   2144.29
2347167796   2093.41
1503960366   1945.94
2873212765   1906.90
8792009665   1901.50
3372868164   1900.03
4057192912   1895.72
6391747486   1805.21
5553957443   1767.94
4558609924   1764.34
8253242879   1495.13
1844505072   1471.81
2026352035   1452.91
2320127002   1444.88
1624580081   1384.22
3977333714   1267.41
2891001357   1120.00

### Análise de sobrepeso nos usuários do fitbit
* 35 ids tem dados sobre peso
* 7 ids do fit estão acima do peso, e apenas 1 delas tem obesidade grau 2 e apenas 1 tem obesidade grau 3
* comparando weightlog e daily activity, 3 usuários acima do peso  tem inputs com total step zerado ,mas não todos os inputs de total steps zerados

In [46]:
overweight_ids = weightlog.loc[weightlog['BMI'] > 25, 'Id'].unique()

In [47]:
weightlog[weightlog['Id'].isin(overweight_ids)]

Id                   Date    WeightKg  WeightPounds  Fat  \
1   1927972279   4/10/2016 6:33:26 PM  129.600006    285.719105  NaN   
5   2891001357   4/5/2016 11:59:59 PM   88.400002    194.888643  NaN   
6   4445114986  3/30/2016 11:59:59 PM   92.400002    203.707134  NaN   
7   4558609924   4/8/2016 11:59:59 PM   69.400002    153.000813  NaN   
8   4702921684   4/4/2016 11:59:59 PM   99.699997    219.800869  NaN   
23  8253242879   4/7/2016 11:59:59 PM   75.599998    166.669467  NaN   
24  8877689391    4/1/2016 6:49:40 AM   85.500000    188.495234  NaN   
25  8877689391    4/4/2016 6:53:43 AM   86.599998    190.920316  NaN   
26  8877689391    4/5/2016 6:40:15 AM   86.000000    189.597545  NaN   
27  8877689391    4/6/2016 6:49:31 AM   86.300003    190.258939  NaN   
28  8877689391    4/7/2016 6:15:08 AM   85.099998    187.613382  NaN   
29  8877689391    4/8/2016 6:39:44 AM   85.000000    187.392923  NaN   
30  8877689391    4/9/2016 8:06:12 AM   85.400002    188.274775  NaN   
31  8877689391   4/11/2016 6:58:09 AM   86.099998    189.818004  NaN   
32  8877689391   4/12/2016 6:47:11 AM   85.800003    189.156628  NaN   

          BMI  IsManualReport          LogId  
1   46.169998           False  1460313206000  
5   25.030001            True  1459900799000  
6   35.009998            True  1459382399000  
7   27.139999            True  1460159999000  
8   26.110001            True  1459814399000  
23  29.549999            True  1460073599000  
24  25.610001           False  1459493380000  
25  25.940001           False  1459752823000  
26  25.760000           False  1459838415000  
27  25.830000           False  1459925371000  
28  25.490000           False  1460009708000  
29  25.440001           False  1460097584000  
30  25.559999           False  1460189172000  
31  25.790001           False  1460357889000  
32  25.680000           False  1460443631000

Apenas 7 id únicos pessoas dos dados usuárias do fit estão acima do peso, e apenas 1 delas tem obesidade grau 2 e apenas 1 tem obesidade grau 3

### Analisar os dados desse grupo amostral nos outros csv's

In [ ]:
d_activity[(d_activity['TotalSteps']== 0) & (d_activity['Id'].isin(overweight_ids))] #3 usuários acima do peso  tem inputs com total step zerado

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
135  2891001357    3/29/2016           0            0.0              0.0   
136  2891001357    3/30/2016           0            0.0              0.0   
137  2891001357    3/31/2016           0            0.0              0.0   
138  2891001357     4/1/2016           0            0.0              0.0   
140  2891001357     4/3/2016           0            0.0              0.0   
141  2891001357     4/4/2016           0            0.0              0.0   
290  4702921684    4/12/2016           0            0.0              0.0   
401  8253242879     4/1/2016           0            0.0              0.0   
402  8253242879     4/2/2016           0            0.0              0.0   
403  8253242879     4/3/2016           0            0.0              0.0   
404  8253242879     4/4/2016           0            0.0              0.0   
405  8253242879     4/5/2016           0            0.0              0.0   
412  8253242879    4/12/2016           0            0.0              0.0   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
135                  0.000000                 0.0                       0.0   
136                  0.000000                 0.0                       0.0   
137                  0.000000                 0.0                       0.0   
138                  4.828032                 0.0                       0.0   
140                  0.000000                 0.0                       0.0   
141                  0.000000                 0.0                       0.0   
290                  0.000000                 0.0                       0.0   
401                  0.000000                 0.0                       0.0   
402                  0.000000                 0.0                       0.0   
403                  0.000000                 0.0                       0.0   
404                  0.000000                 0.0                       0.0   
405                  0.000000                 0.0                       0.0   
412                  0.000000                 0.0                       0.0   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
135                  0.0                      0.0                  0   
136                  0.0                      0.0                  0   
137                  0.0                      0.0                  0   
138                  0.0                      0.0                  0   
140                  0.0                      0.0                  0   
141                  0.0                      0.0                  0   
290                  0.0                      0.0                  0   
401                  0.0                      0.0                  0   
402                  0.0                      0.0                  0   
403                  0.0                      0.0                  0   
404                  0.0                      0.0                  0   
405                  0.0                      0.0                  0   
412                  0.0                      0.0                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
135                    0                     0              1440      1920  
136                    0                     0              1440      1920  
137                    0                     0              1440      1920  
138                  660                     0               780      4562  
140                    0                     0              1440      1920  
141                    0                     0              1440      1920  
290                    0                     0              1440         0  
401                    0                     0              1440      1429  
402                    0                     0              1440      1429  
403                    0                     0              14

# insights até aqui